pkg/ingester/ingester.go

```diff
@@ -630,7 +640,18 @@ func (i *Ingester) starting(ctx context.Context) (err error) {
 	}
 
 	i.subservicesAfterIngesterRingLifecycler, err = createManagerThenStartAndAwaitHealthy(ctx, servs...)
-	return errors.Wrap(err, "failed to start ingester subservices after ingester ring lifecycler")
+	if err != nil {
+		return errors.Wrap(err, "failed to start ingester subservices after ingester ring lifecycler")
+	}
+
+	if i.cfg.CircuitBreakerConfig.InitialDelay == 0 {
+		i.circuitBreaker.setActive()
+	} else {
+		time.AfterFunc(i.cfg.CircuitBreakerConfig.InitialDelay, func() {
+			i.circuitBreaker.setActive()
+		})
+	}
```

Comment:
Same here, IMO, this logic should be inside of `newCircuitBreaker`.
This method shouldn't read circuit breaker's config and take decisions on it, it should just call `i.circuitBreaker.onStarting()` or something similar.

Reply:
Oh, yes, sorry, I wrote half comment and then realised that this wasn't part of `newIngester` anymore. I agree, it shouldn't be in `newCircuitBreaker`, but should be in a method of circuit breaker (as I stated in the second part of my comment 😂 )

Reply:
I have moved the logic in the `circuitBreaker.activate()` method: if `cfg.InitialDelay` is 0, the circuit breaker gets activated immediately. Otherwise it gets activated after the initial delay.